maps/hazards.go

```diff
@@ -565,20 +565,70 @@ Each river has one or two 1-square "bridges" over them`,
 		Author:     "Battlesnake",
 		Version:    1,
 		MinPlayers: 1,
-		MaxPlayers: 8,
+		MaxPlayers: 12,
 		BoardSizes: FixedSizes(Dimensions{11, 11}, Dimensions{19, 19}, Dimensions{25, 25}),
 	}
 }
 
 func (m RiverAndBridgesHazardsMap) SetupBoard(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
-	if err := (StandardMap{}).SetupBoard(lastBoardState, settings, editor); err != nil {
+	width := lastBoardState.Width
+	height := lastBoardState.Height
+	startPositions, ok := riversAndBridgesStartPositions[rules.Point{X: width, Y: height}]
+	if !ok {
+		return rules.RulesetError("board size is not supported by this map")
+	}
+
+	numSnakes := len(lastBoardState.Snakes)
+	if numSnakes == 0 {
+		return rules.RulesetError("too few snakes - at least one snake must be present")
+	}
+
+	maxSnakes := len(startPositions)
+	if maxSnakes < numSnakes {
+		return rules.ErrorTooManySnakes
+	}
+
+	rand := settings.GetRand(0)
+
+	snakeIDs := make([]string, 0, len(lastBoardState.Snakes))
+	for _, snake := range lastBoardState.Snakes {
+		snakeIDs = append(snakeIDs, snake.ID)
+	}
+
+	tempBoardState := rules.NewBoardState(width, height)
+	tempBoardState.Snakes = make([]rules.Snake, len(snakeIDs))
+
+	for i := 0; i < len(snakeIDs); i++ {
+		tempBoardState.Snakes[i] = rules.Snake{
+			ID:     snakeIDs[i],
+			Health: rules.SnakeMaxHealth,
+		}
+	}
+	err := rules.PlaceSnakesAtPositions(tempBoardState, startPositions)
+	if err != nil {
+		return err
+	}
```

Comment:
This doesn't randomize the starting positions, right? I think we want them to be randomized.
The `battlesnake` CLI seems to accidentally cause this to happen by storing the snakes in a map, which randomly orders them.

Reply:
I think randomly ordering makes sense. I was expecting it to be the case, but the Python code I was translating didn't have any and I strictly followed its approach. But maybe we should change that? It would be pretty simple to add some randomness.